cmd/downloads/config/downloads_config.go

```
@@ -397,9 +388,15 @@ func (c *DownloadsServerConfig) CreateArchivesInBackground() {
 func (c *DownloadsServerConfig) Handler() http.Handler {
 	fs := http.FileServer(http.Dir(c.TempDir))
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ext := filepath.Ext(r.URL.Path)
-		if ext == ".tar" || ext == ".zip" {
-			<-c.archivesReady
+		if strings.HasSuffix(r.URL.Path, ".zip") || strings.HasSuffix(r.URL.Path, ".tar.gz") {
+			select {
+			case <-r.Context().Done():
+				w.WriteHeader(http.StatusServiceUnavailable)
+				fmt.Fprintln(w, "the server is not ready yet. try again later.")
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintln` is not checked

(errcheck)